tests/console/yast2_settings.pm

```diff
@@ -0,0 +1,85 @@
+# coding: utf-8
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: YaST writes config to 70-yast.conf and detects conflicts
+# - Install yast2-network
+# - verify writing config to the correct file
+# - ensure conflict detected when custom confing with high higher priority exist
+# - validate that YaST applies changes to the system. bsc#1167234
+# Maintainer: QA SLE YaST team <qa-sle-yast@suse.de>
+
+use base "y2_module_consoletest";
+use strict;
+use warnings;
+use testapi;
+use utils;
+use y2lan_restart_common;
+
+my $yastconf              = '/etc/sysctl.d/70-yast.conf';
+my $customconf_in_etc     = '/etc/sysctl.d/90-custom.conf';
+my $customconf_in_usr_lib = '/usr/lib/sysctl.d/90-custom.conf';
+
+sub verify_write_config {
+    open_yast2_routing();
+    send_key 'spc';
+    assert_screen "ipv4_forwarding_enabled";
+    send_key 'alt-n';
+    wait_serial("yast2-routing-status-0", 180) || die "'yast2-routing' didn't finish";
+    assert_script_run("sysctl net.ipv4.ip_forward | grep 1");
+}
+
+sub ensure_conflict_detected {
+    _create_conflicts();
+    open_yast2_routing();
+    send_key 'spc';
+    assert_screen "ipv4_forwarding_disabled";
+    send_key 'alt-n', wait_screen_change => 1;
+    assert_screen "sysctl_ip_forward_conflict";
+    send_key 'alt-o';
+    wait_serial("yast2-routing-status-0", 180) || die "'yast2-routing' didn't finish";
+}
+
+sub validate_changes_applied_to_system {
+    assert_script_run("cat $yastconf |grep 'net.ipv4.ip_forward = 1'");
+    assert_script_run("sysctl net.ipv4.ip_forward | grep 1");
+    assert_script_run("cat /proc/sys/net/ipv4/ip_forward | grep 1", fail_message => "ip_forward not applied to system bsc#1167234");
+}
+
+sub _create_conflicts {
+    # Files with higher priority that 70-yast.conf.
+    # But /etc has precedence over /usr/lib, thus second one should not
+    # cause conflict
+    my @other_confs = ($customconf_in_etc, $customconf_in_usr_lib);
+    foreach my $conf (@other_confs) {
+        assert_script_run("touch $conf");
+        my $create_other_conf = 'cat > ' . $conf . ' << EOF
+net.ipv4.ip_forward=1
```

Comment:
the same settings is used over and over, maybe a constant at the beginning of the file would help.

Reply:
As this is going to be used only inside this block of code i cant really see why it should be a constant. the scope is only inside the block. i could come up with some alternative but i believe it is unnecessary code to implement.

Reply:
`net.ipv4.ip_forward` is repeated all over the file not only in this block.

Reply:
ok. i didnt get that you were talking about this. i will take care